### PR TITLE
SOC-1971 - Include web contributions in DiscussionsLog

### PR DESCRIPTION
**Pull request by katanka**
This switches the discussions log to look at contribution info logged from the service instead of only mobile apps.

### CODE REVIEW

File: extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLog.i18n.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	'discussionslog-view-logs' => 'View Logs',
 	'discussionslog-no-user-match-error' => 'Provided username did not match any user',
 	'discussionslog-ip-invalid-error' => 'Provided IP address is invalid.',
-	'discussionslog-no-mobile-activity-error' => 'No mobile app activity by $1 in the past two weeks!',
+	'discussionslog-no-activity-error' => 'No contributions by $1 in the past two weeks!',
 	'discussionslog-no-ip-activity-error' => 'No recent activity from IP address $1 found.',
```
Comment (katanka) [inline extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLog.i18n.php:16]: Reminder to add new tag.

Comment (armonr) [inline extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLog.i18n.php:16]: @katanka As discussed offline, better approach is to create a new translation key and then remove the old one. Also we need a LOC ticket.

File: extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php [hunk 2]
```diff
@@ -297,4 +297,25 @@ private function checkAccess() {
 			throw new \PermissionsError( self::DISCUSSIONS_LOG_ACTION );
 		}
 	}
+
+	private function getLocationFromIP( $ip, array &$ipCache ) {
+		if (array_key_exists($ip, $ipCache)) {
```
Comment (armonr) [inline extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php]: Simpler:

``` php
if ( isset( $ipCache[$ip] ) ) {
```

File: extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php [hunk 3]
```diff
@@ -297,4 +297,25 @@ private function checkAccess() {
...
+			return $ipCache[$ip];
+		}
+
+		$details = json_decode(file_get_contents("http://ipinfo.io/{$ip}/json"));
```
Comment (armonr) [inline extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php]: What if this site goes down? :)
Try looking at GeoIP library, possibly this https://github.com/Wikia/app/blob/master/lib/vendor/Net/GeoIP.php#L661

File: extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php [hunk 4]
```diff
@@ -297,4 +297,25 @@ private function checkAccess() {
...
+		$location_arr = array($details->country);
```
Comment (garthwebb) [inline extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php]: Use the new style array syntax:

``` php
$location_arr = [ $details->country ];
```

File: extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php [hunk 5]
```diff
@@ -297,4 +297,25 @@ private function checkAccess() {
...
+
+		if ( $details->region ) {
+			array_unshift($location_arr, $details->region);
+		}
+
+		if ( $details->city ) {
+			array_unshift($location_arr, $details->city);
+		}
```
Comment (garthwebb) [inline extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php]: So it looks like you're using `array_unshift` to get these in a City, Region, Country order.  However, unshifting wouldn't be necessary if you added these in the order you want to begin with.  Then you could use the standard PHP array push syntax, e.g.:

``` php
$location_arr[] = $details->city
```

File: extensions/wikia/SpecialDiscussionsLog/templates/SpecialDiscussionsLog_userLog.mustache [hunk 6]
```diff
@@ -20,16 +20,59 @@
...
+					<td class="location">Loading...</td>
 					<td>{{language}}</td>
-					<td>{{userAgent}}</td>
+					<td class="useragent">{{userAgent}}</td>
 					<td>{{app}}</td>
 					<td>{{timestamp}}</td>
 				</tr>
 			{{/userLogRecords}}
 		</table>
+        <script>
```
Comment (armonr) [inline extensions/wikia/SpecialDiscussionsLog/templates/SpecialDiscussionsLog_userLog.mustache]: Please make a separate JS file for this and then add it to the setup file.

**Comment by armonr** [comment]
LGTM

**Pull request merged**